### PR TITLE
De-Duplicate message types.

### PR DESCRIPTION
**Pull request by icholy**
#43 @adrianmo @sthorshaug I took a stab at it. Let me know what you think.

### CODE REVIEW

**Comment by icholy** [comment]
Adding more talkers is the easy part, we need to figure out if the breaking changes here are worth it.

**Comment by icholy** [comment]
@adrianmo do you think it's worth trying to maintain backwards compatibility? or just bump the major version?

``` go
type (
	GLGSV     = GSV
	GLGSVInfo = GSVInfo
	GNGGA     = GGA
	GNGNS     = GNS
	GNRMC     = RMC
	GPGGA     = GGA
	GPGLL     = GLL
	GPGSA     = GSA
	GPGSV     = GSV
	GPGSVInfo = GSVInfo
	GPHDT     = HDT
	GPRMC     = RMC
	GPVTG     = VTG
	GPZDA     = ZDA
)
```

**Comment by icholy** [comment]
I've added `deprecated.go` which makes this backwards compatible with the last release. I've also included all the old tests in `deprecated_test.go` which all pass without modification.

File: sentence.go [hunk 1]
```diff
@@ -57,14 +60,28 @@ func parseSentence(raw string) (BaseSentence, error) {
...
+		Talker:   talker,
+		Type:     typ,
 		Fields:   fields[1:],
 		Checksum: checksumRaw,
 		Raw:      raw,
 	}, nil
 }
 
+// parsePrefix takes the first field and splits it into a talker id and data type.
+// If it's a proprietary sentence, then return the entire prefix as the type.
```
Comment (adrianmo) [inline sentence.go]: > If it's a proprietary sentence, then return the entire prefix as the type

It looks like the type does not contain the initial `P`, so the type is not the entire prefix.

Comment (icholy) [reply sentence.go]: @adrianmo good catch. This was the original behaviour, but I ended up changing it.

**Comment by icholy** [comment]
@adrianmo @sthorshaug I wanted to be as conservative as possible with the talker id's but it seems like you both want to remove the limits (eg `AssertTalkers`). I'll drop those constraints.

**Comment by sthorshaug** [comment]
@icholy I'm impressed with your dedication for this library, and I'm certain this is a good solution!

**Comment by adrianmo** [comment]
@icholy can you please update the README file to remove the talker ID from the list of supported sentences? Thanks!

**Comment by icholy** [comment]
@sthorshaug I use this in production, so it's a win-win :+1:

**Comment by icholy** [comment]
@kortschak this was [acknowledged](https://github.com/adrianmo/go-nmea/pull/44#issuecomment-449064964)

**Comment by kortschak** [comment]
OK, so try that in a type switch. Now you have a failure to compile because of a duplicated case. I don't think this is a good idea, being a false claim as it is.

**Comment by icholy** [comment]
@kortschak ah, I didn't think of that case. Are you proposing an alternative solution?

**Comment by kortschak** [comment]
Given the current behaviour, I'd suggest you just break people once and remove the aliases now. This use of them is not what they were designed for.

**Comment by icholy** [comment]
@adrianmo what do you think?

**Comment by kortschak** [comment]
Sorry, I missed this because of edition not being a trigger for notification.

The type alias talks about how type alias should be used for refactoring, which is what I presume you are referring to. The important part of that document as related to this is

1. Add new API, *interchangeable with old API*.
2. Convert uses of old API to new API.
3. Remove old API.

The important part of the process is the highlighted section of 1. This is not interchangeable.

**Comment by icholy** [comment]
@kortschak yeah I missed that when making this change. None of my client code had the same aliased type in the type switch. Either way, I don't think we'll be rolling back this change. If you're proposing a solution, I'm all ears.

**Comment by kortschak** [comment]
The proposal is to delete `deprecated{,_test}.go`.

**Comment by icholy** [comment]
@kortschak that is the intention.

**Comment by kortschak** [comment]
The point I was making was that should happen now since the claimed benefit of a deprecation period does not actually exist.

**Comment by icholy** [comment]
@kortschak open a PR

**Pull request merged**